editor/js/Sidebar.Geometry.LatheGeometry.js

```diff
@@ -57,10 +57,37 @@ function GeometryParametersPanel( editor, object ) {
 
 	//
 
+	function isPointsIdentical( newPoints ) {
+
+		const oldPoints = points.getValue();
+
+		if ( oldPoints.length !== newPoints.length ) return false;
+
+		for ( let i = 0; i < oldPoints.length; i ++ ) {
+
+			const oldPoint = oldPoints[ i ];
+			const newPoint = newPoints[ i ];
+
+			if ( oldPoint.equals( newPoint ) === false ) return false;
+
+		}
+
+		return true;
+
+	}
+
 	function refreshUI() {
 
 		const parameters = object.geometry.parameters;
 
+		if ( isPointsIdentical( parameters.points ) === false ) {
```

Comment:
How about implementing the equals checks in `setValue()` instead`? In this way, you can just execute `points.setValue( parameters.points );` in `refreshUI()`.

In `setValue()`, the UI is only build when the points set is different. This makes it also possible to control whether `update()` is called or not. So you don't have to change the `onChange()` handler.

Reply:
ok.